component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java

```diff
@@ -315,9 +315,13 @@ public Response getUsers(@Context UriInfo uriInfo,
         identities = getOnlineIdentities(userStateService, userId, limit);
       }
     } else {
+      Identity target = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, userId);
       ProfileFilter filter = new ProfileFilter();
       filter.setName(q == null || q.isEmpty() ? "" : q);
       filter.setEnabled(!isDisabled);
+      if (target != null && filter.getName().equals(q)) {
+        filter.setExcludedIdentityList(Collections.singletonList(target));
+      }
       if (!isDisabled) {
```

Comment:
@ahamdi The getUsers function (and the restService in general) provide the list of users which match all parameters. There is no information about excluding the current user.
I'm afraid that excluding the current user in the rest service can lead to problems in other parts of the platform, where the current user is needed.

If this task is too complex to fix, we can postpone it to 6.4.1.

I think we should take reflexion time here

Reply:
@rdenarie Yes very possible for 6.4.0

Reply:
ok, so go with a new parameter in the function. 

I keep the possibility to push back in 6.4.1 if not fully finished at the end of the week